components/PyTorch/Create_PyTorch_Model_Archive/component.yaml

```diff
@@ -0,0 +1,39 @@
+name: Create PyTorch Model Archive
+inputs:
+- {name: Model, type: PyTorchScriptModule}
+- {name: Model name, type: String, default: model}
+- {name: Model version, type: String, default: "1.0"}
+- {name: Handler, type: PythonCode, description: "See https://github.com/pytorch/serve/blob/master/docs/custom_service.md"}
+outputs:
+- {name: Model archive, type: PyTorchModelArchive}
+metadata:
+  annotations:
+    author: Alexey Volkov <alexey.volkov@ark-kun.com>
+implementation:
+  container:
+    image: pytorch/torchserve:0.3.0-cpu
+    command:
+    - bash
+    - -exc
+    - |
+      model_path=$0
+      handler_path=$1
+      model_name=$2
+      model_version=$3
+      output_model_archive_path=$4
+      
+      mkdir -p "$(dirname "$output_model_archive_path")"
+
+      # torch-model-archiver needs the handler to have .py extension
+      cp "$handler_path" handler.py
+      torch-model-archiver --model-name "$model_name" --version "$model_version" --serialized-file "$model_path" --handler handler.py
+      
+      # torch-model-archiver does not allow specifying the output path, but always writes to "${model_name}.<format>"
```

Comment:
I'm no expert, but I believe the `--export-path` arg lets you do this.  
There are a few other possible args as well.  Should the component support all of them as options?
```
torch-model-archiver [-h] --model-name MODEL_NAME [--serialized-file SERIALIZED_FILE] [--model-file MODEL_FILE] --handler HANDLER [--extra-files EXTRA_FILES] [--runtime {python,python2,python3}]
                            [--export-path EXPORT_PATH] [--archive-format {tgz,no-archive,default}] [-f] -v VERSION [-r REQUIREMENTS_FILE]
```

Reply:
I found that the --export-path does not change this. It just changes the directory.
https://github.com/pytorch/serve/blob/f0da8d2714628cb5de0c6eaa3ac0448a6b226640/model-archiver/model_archiver/model_packaging_utils.py#L48

>There are a few other possible args as well. Should the component support all of them as options?

Generally yes, but:
* Some are not relevant to containerized components (e.g. `--export-path`) - the user does not care where some intermediate file is placed inside a short-lived container.
* Some parameters are fixed due to the component purpose. This component explicitly converts to `MAR` format, so the `--archive-format` is set to default.

Reply:
I'll try adding the `-r` option later.